test/test-execute/exec-temporaryfilesystem-options.service

```diff
@@ -0,0 +1,11 @@
+[Unit]
+Description=Test for TemporaryFileSystem with mount options
+
+[Service]
+Type=oneshot
+
+# Check /proc/self/mountinfo
+ExecStart=/bin/sh -c 'test $$(awk \'$$5 == "/var" { print $$6 }\' /proc/self/mountinfo) = "ro,nodev,relatime"'
+ExecStart=/bin/sh -c 'test $$(awk \'$$5 == "/var" { print $$11 }\' /proc/self/mountinfo) = "ro,mode=700"'
```

Comment:
I'm wondering why `test-execute` didn't fail on `Fedora`. There is usually `seclabel` in the 11th field of `/proc/self/mountinfo`, which makes it impossible for `ro,mode=700` to pass the check.

Reply:
```sh
exec-temporaryfilesystem-options.service: Passing 0 fds to service
exec-temporaryfilesystem-options.service: About to execute: /bin/sh -c 'test $$(awk '$$5 == "/var" { print $$6 }' /proc/self/mountinfo) = "ro,nodev,relatime"'
exec-temporaryfilesystem-options.service: Forked /bin/sh as 10524
Followed symlinks /run/systemd/unit-root/var → /run/systemd/unit-root/var.
Applying namespace mount on /run/systemd/unit-root/var
exec-temporaryfilesystem-options.service: Executing: /bin/sh -c 'test $(awk '$5 == "/var" { print $6 }' /proc/self/mountinfo) = "ro,nodev,relatime"'
Received SIGCHLD from PID 10524 (sh).
Child 10524 (sh) died (code=exited, status=0/SUCCESS)
exec-temporaryfilesystem-options.service: Child 10524 belongs to exec-temporaryfilesystem-options.service.
exec-temporaryfilesystem-options.service: Main process exited, code=exited, status=0/SUCCESS
exec-temporaryfilesystem-options.service: Running next main command for state start.
exec-temporaryfilesystem-options.service: Passing 0 fds to service
exec-temporaryfilesystem-options.service: About to execute: /bin/sh -c 'test $$(awk '$$5 == "/var" { print $$11 }' /proc/self/mountinfo) = "ro,mode=700"'
exec-temporaryfilesystem-options.service: Forked /bin/sh as 10527
Followed symlinks /run/systemd/unit-root/var → /run/systemd/unit-root/var.
Applying namespace mount on /run/systemd/unit-root/var
exec-temporaryfilesystem-options.service: Executing: /bin/sh -c 'test $(awk '$5 == "/var" { print $11 }' /proc/self/mountinfo) = "ro,mode=700"'
Received SIGCHLD from PID 10527 (sh).
Child 10527 (sh) died (code=exited, status=1/FAILURE)
exec-temporaryfilesystem-options.service: Child 10527 belongs to exec-temporaryfilesystem-options.service.
exec-temporaryfilesystem-options.service: Main process exited, code=exited, status=1/FAILURE
exec-temporaryfilesystem-options.service: Failed with result 'exit-code'.
exec-temporaryfilesystem-options.service: Changed start -> failed
exec-temporaryfilesystem-options.service: Unit entered failed state.
	PID: 10527
	Start Timestamp: Wed 2018-02-21 09:24:08 UTC
	Exit Timestamp: Wed 2018-02-21 09:24:08 UTC
	Exit Code: exited
	Exit Status: 1
Assertion 'service->main_exec_status.status == status_expected' failed at ../src/test/test-execute.c:74, function check(). Aborting.
Aborted
```

Reply:
Ouch!

Reply:
looks good. Can I include this in a PR that makes `ReadWritePaths=` can work with `TemporaryFileSystem=`.

Reply:
You are welcome to take the above code, I just want to make sure the tests not left broken (on Fedora) for too long.  If you think the ReadWritePaths= will be quick to merge then I guess you can include them together.

Reply:
I just changed my mind. This is now pushed as #8241. Please take a look.

Reply:
@sourcejedi, @yuwata thank you for looking at this. I've already fixed the test locally, but I hesitated to show it because, well, here it is:
```diff
diff --git a/test/test-execute/exec-temporaryfilesystem-options.service b/test/test-execute/exec-temporaryfilesystem-options.service
index 1d5d76c81..d63f66fd2 100644
--- a/test/test-execute/exec-temporaryfilesystem-options.service
+++ b/test/test-execute/exec-temporaryfilesystem-options.service
@@ -5,7 +5,13 @@ Description=Test for TemporaryFileSystem with mount options
 Type=oneshot

 # Check /proc/self/mountinfo
-ExecStart=/bin/sh -c 'test $$(awk \'$$5 == "/var" { print $$6 }\' /proc/self/mountinfo) = "ro,nodev,relatime"'
-ExecStart=/bin/sh -c 'test $$(awk \'$$5 == "/var" { print $$11 }\' /proc/self/mountinfo) = "ro,mode=700"'
+ExecStart=/usr/bin/perl -alne 'if ($$F[4] eq "/var") { \
+  my %%a = map { $$_ => 1 } split(",", $$F[10]); \
+  if (!@a{ro,"mode=700"}) {next}; \
+  my %%b = map { $$_ => 1 } split(",", $$F[5]); \
+  if (@b{ro,nodev,relatime}) {$$found=1; exit}; \
+} \
+END {exit($$found ? 0 : 1)}' \
+/proc/self/mountinfo

 TemporaryFileSystem=/var:ro,mode=0700,nostrictatime
```
Don't ask me how and why it works. I don't know :-)